Convert BigInteger to Int64 when possible from Python

aparajit-pratap opened the pull request:
### Purpose

https://autodesk.slack.com/archives/CED3BF6C9/p1571942488031200
![image](https://user-images.githubusercontent.com/5710686/68160134-130af680-ff21-11e9-8302-7354dfccc950.png)

JIRA: https://jira.autodesk.com/browse/DYN-2262

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo 

### FYIs

@Dewb

aparajit-pratap commented on src/DynamoUtilities/DataMarshaler.cs:
```diff
@@ -86,7 +87,20 @@ public object Marshal(object obj)
             if (!applicable.Any())
-                return obj;
+            {
+                if (!(obj is BigInteger bigInt)) return obj;
```
C# 7.0 syntax seems to be breaking the engops build.

Dewb commented:
Excellent! 

Any reason this has to be done inside the generic marshaller execution rather than by defining a custom marshaler for `BigInteger`?

Dewb commented:
Looks like we have custom `DataMarshaler`s for Python node inputs, but not outputs. This new code could definitely go here: https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/DSIronPython/IronPythonEvaluator.cs

The `DataMarshaler` class isn't used anywhere aside from the Python node so I would also support moving it out of `DynamoUtilities` and into the Python project, to make its purpose more clear. (Doesn't have to be part of this PR.)

aparajit-pratap commented:
> Looks like we have custom `DataMarshaler`s for Python node inputs, but not outputs. This new code could definitely go here: https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/DSIronPython/IronPythonEvaluator.cs

The `DataMarshaler` class is used for both input and output marshaling as seen here: https://github.com/DynamoDS/Dynamo/blob/8fb3b6829c9bcd8c01ee4ef7e880aa8185bf2cd0/src/Libraries/DSIronPython/IronPythonEvaluator.cs#L114 and here: https://github.com/DynamoDS/Dynamo/blob/8fb3b6829c9bcd8c01ee4ef7e880aa8185bf2cd0/src/Libraries/DSIronPython/IronPythonEvaluator.cs#L134, however, I agree that the marshaling for `BigInteger` can be specific to the output marshaling.

mjkkirschner commented:
@aparajit-pratap would you mind filing this internally?

aparajit-pratap commented:
Added unit test, addressed review comments about registering the `BigInteger` marshaling logic only for node output (output marshalers are added only in the `DataMarshaler` constructor, while input marshalers are added when first initializing the `IronPythonEvaluator` property). 

I haven't moved `DataMarshaler` to the Python project just yet as @mjkkirschner had some other thoughts on this. Leaving it open for discussion.

QilongTang commented on src/DynamoUtilities/DynamoUtilities.csproj:
```diff
@@ -49,6 +49,7 @@
     <Reference Include="System" />
     <Reference Include="System.configuration" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Numerics" />
```
Do we need this reference?

aparajit-pratap replied:
Yes, `BigInteger` is defined in `System.Numerics`.

QilongTang commented:
LGTM with one comments

Dewb commented:
FYI the IronPython issue that prompted this is supposedly fixed in IronPython 2.7.9.

mjkkirschner commented:
hmm - so this test should have passed already - since we are using ironPython 2.7.9.

@aparajit-pratap do we now include system.numerics in our shipped binaries or this is loaded from the GAC?

It will be useful to know as we should communicate this in the release notes of 2.5

aparajit-pratap commented:
I reverted my change leaving the test to test locally. It didn't pass?

mjkkirschner commented:
@aparajit-pratap your test also includes a big integer check (last type in your list) - perhaps that gets marshalled to long now in all cases?

mjkkirschner commented:
here's Mike's issue:
https://github.com/IronLanguages/ironpython2/issues/681

aparajit-pratap commented:
> @aparajit-pratap your test also includes a big integer check (last type in your list) - perhaps that gets marshalled to long now in all cases?

This is the output I'm seeing before my change:
![image](https://user-images.githubusercontent.com/5710686/68730271-332e5b80-059a-11ea-9a53-33c2b144899f.png)

aparajit-pratap commented:
> @aparajit-pratap do we now include system.numerics in our shipped binaries or this is loaded from the GAC?

It is loaded from GAC and referenced from `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7\`
